__tests__/app.test.js

```diff
@@ -0,0 +1,35 @@
+const app = require('../app');
+const seed = require("../db/seeds/seed");
+const request = require('supertest');
+const db = require('../db/connection');
+const data = require("../db/data/test-data");
+
+beforeEach(() => seed(data));
+afterAll(() => db.end());
+
+describe("GET/api/topics", () => {
+    it("returns a status code of 200", () => {
+        return request(app)
+        .get("/api/topics")
+        .expect(200);
+    });
+    it("should return an object with a key of topics", () => {
+        return request(app)
+        .get("/api/topics")
+        .then(({body}) => {
+            expect(body).toHaveProperty("topics");
+        });
+    });
+    it("should have a 'slug' and 'description' key in each returned object within the topics array", () => {
+        return request(app)
+        .get("/api/topics")
+        .then(({body}) => {
+            body.topics.forEach(topic => {
+                expect(topic).toEqual(expect.objectContaining({
+                    slug: expect.any(String),
+                    description: expect.any(String)
+                }));
+            })
+        });
```

Comment:
Nice structure of the `forEach` but...what would happen if the `body.topics` array was empty? Would this test pass or fail?

Reply:
It would still pass because the tests inside the forEach wouldn't run. So the pass wouldn't be for the right reasons. I've implemented a workaround, which should still pass whether the array is empty or not, but this time for the expected reasons.